mgmt/rest/rest_func_test.go

```diff
@@ -582,6 +583,12 @@ func TestPluginRestCalls(t *testing.T) {
 				cfg := getDefaultMockConfig()
 				err := cfgfile.Read("../../examples/configs/snap-config-sample.json", &cfg, MOCK_CONSTRAINTS)
 				So(err, ShouldBeNil)
+				if len(SNAP_AUTODISCOVER_PATH) == 0 {
+					if len(SNAP_PATH) != 0 {
+						SNAP_AUTODISCOVER_PATH = fmt.Sprintf("%s/plugin", SNAP_PATH)
+					}
+				}
+				cfg.Control.AutoDiscoverPath = SNAP_AUTODISCOVER_PATH
```

Comment:
Thank you for explaining. I knew there was a reason I just didn't see it. I understand why the failures were happening now though. The previous behavior was that auto-loading never happened here because this calls control directly and autoloading only happened in snapd. 

I'm not sure I understand the rest about using it to test plugins loading from another location. What is the rationale behind that?